Task2/usersAPI.ts

```diff
@@ -0,0 +1,85 @@
+import express from 'express';
+import uuidv4 from 'uuid/v4';
+
+import { usersCollection } from './usersCollection';
+import validateUser from './checkValidation';
+
+const app: express.Application = express();
+const router = express.Router();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+app.use('/users', router);
+
+function getAutoSuggestUsers (loginSubstringin:string, limit: string):any {
+	if (!loginSubstringin || loginSubstringin.length === 0) return usersCollection;
+	loginSubstringin = loginSubstringin.toLowerCase();
+	const numLimit: number = limit ? +limit : usersCollection.length;
+
+	const filteredAndSortedUsersCollection = usersCollection.filter(user => {
+		return user.login.startsWith(loginSubstringin);
+	}).sort((userA, userB) => {
+		return userA.login < userB.login ? -1 : 1;
+	});
+	const limitCollection = Math.min(
+		filteredAndSortedUsersCollection.length, 
+		numLimit);				
+	return filteredAndSortedUsersCollection.slice(0, limitCollection);
+	
+}
+
+router.route('/')
+	.get(function (req, res, next) {
+		if(Object.keys(req.query).length !== 0){
+			const usersCollection = getAutoSuggestUsers(req.query.login, req.query.limit);
+			(usersCollection && usersCollection.length) > 0
+				? res.json(usersCollection)
+				: res.status(404).send('No one user was found :(');
+		} else {
+			res.json(usersCollection);
```

Comment:
Returning of all collection is OK for homework. But in real life, it can be 1000000 users in the database and you don't want to return all the collection at once =)
So, when you have to operate with a large collection, consider using pagination.